### PR TITLE
dynamic variable to stop mangling

### PR DESCRIPTION
**Pull request by piranha**
So obviously some tests would be nice, but I'm not sure how to write them. I can make current ones fail with `(binding [abracad.avro.util/*mangle-names* false] ...)` (on having "-" as illegal char), but what's the best way to test for *not mangling*?

### CODE REVIEW

**Comment by piranha** [comment]
Oh, and also macro - what do you mean about that? Not sure I understand.

File: src/clojure/abracad/avro/util.clj [hunk 1]
```diff
@@ -3,6 +3,8 @@
   {:private true}
   (:import [org.apache.avro Schema$Field]))
 
+(def ^:dynamic *mangle-names* true)
```
Comment (llasram) [inline src/clojure/abracad/avro/util.clj]: This should have a docstring.

**Comment by llasram** [comment]
I'd definitely like to see some tests.  Unless I'm mistaken, it seems like it should be straightforward.  The Avro spec does not allow field names with `-`, so just need to test that field names with `_` get handled correctly under the 2x2 matrix of no/mangle and de/serialize.

**Comment by llasram** [comment]
Oh, and a macro -- because the dynamic var isn't in the main interface namespace, it might be cleaner to have a macro like `with-mangling` in `abracad.avro` which binds the dynamic var.  I'm on the fence on that one though.

**Comment by piranha** [comment]
Interesting: `unparse-schema` is not doing mangling. Should it?

**Comment by piranha** [comment]
Added some tests, is that close to what you had in mind?

**Comment by llasram** [comment]
The tests look good!

Schemas: I honestly wasn't thinking about schemas at all...  Since `-` is not allowed in Avro field names, any appearance of it in a schema can only mean `_`.  I may make some further changes to ensure mangling always happens during parse, but unmangling definitely shouldn't happen during schema unparse.

If you'll squash the commits, I'll go ahead and merge then cut a new release soon.

**Comment by piranha** [comment]
Here we go! :)

**Comment by llasram** [comment]
And I've now released abracad 0.4.13 with this change.  Thanks!

**Comment by piranha** [comment]
Wow, thanks, that was fast!

**Pull request merged**